tests/models/animals/test_animal_model.py

```diff
@@ -189,3 +197,35 @@ def test_update_method_time_index_argument(
     model.update(time_index=time_index)
 
     assert True
+
+
+def test_calculate_litter_additions(functional_group_list_instance):
+    """Test that litter additions from animal model are calculated correctly."""
+
+    from virtual_rainforest.core.data import Data
+    from virtual_rainforest.core.grid import Grid
+    from virtual_rainforest.models.animals import AnimalModel
+
+    # Create a small data object to work with
+    grid = Grid(cell_nx=2, cell_ny=2)
+    data = Data(grid)
+
+    # Use it to initialise the model
+    model = AnimalModel(data, pint.Quantity("1 week"), functional_group_list_instance)
+
+    # Update the waste pools
+    excess_excrement = [3.5e3, 5.6e4, 5.9e4, 2.3e6]
+    for ind, community in enumerate(model.communities.values()):
+        community.excrement_pool.stored_energy = excess_excrement[ind]
+
+    # Calculate litter additions
+    litter_additions = model.calculate_litter_additions()
+
+    # Check that litter addition pools are as expected
+    assert np.allclose(
+        litter_additions["excess_excrement"], [5e-08, 8e-07, 8.42857e-07, 3.28571e-05]
+    )
+
+    # Check that the function has reset the pools correctly
+    for community in model.communities.values():
+        community.excrement_pool.stored_energy = 0.0
```

Comment:
This doesn't check anything currently.

How about:

```suggestion
    # Check that the function has reset the pools correctly
    assert np.allclose(
        (
            community.excrement_pool.stored_energy
            for community in model.communities.values()
        ),
        0.0,
    )
```